base_report_to_printer/static/src/js/qweb_action_manager.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('base_report_to_printer.print', function (require) {
+    'use strict';
+
+    var ActionManager = require('web.ActionManager');
+    var core = require('web.core');
+    var framework = require('web.framework');
+    var rpc = require('web.rpc');
+
+
+    ActionManager.include({
+        ir_actions_report: function (action, options) {
+            var action_val = _.clone(action);
+            var _t = core._t;
+            var self = this;
+            var _super = this._super;
+
+            if ('report_type' in action_val && action_val.report_type === 'qweb-pdf') {
+                framework.blockUI();
+                rpc.query({
+                    model: 'ir.actions.report',
+                    method: 'print_action_for_report_name',
+                    args: [action_val.report_name]
+                }).then(function (print_action) {
+                    if (print_action && print_action.action_val === 'server') {
+                        framework.unblockUI();
+                        rpc.query({
+                            model: 'ir.actions.report',
+                            method: 'print_document',
+                            args: [action_val.context.active_ids, action_val.report_name],
+                            kwargs: {data: action_val.data || {}},
+                            context: action_val.context || {}
+                        }).then(function () {
+                            self.do_notify(_t('Report'),
+                                _t('Document sent to the printer ') + print_action.printer_name);
```

Comment:
Don't concatenate strings, inerpolate with `%s` and `_.str.sprintf()`, for better translations.